### PR TITLE
MULE-10068: Email Connector - Support connection level headers for SMTP Connections.

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: extensions/email/src/main/java/org/mule/extension/email/api/EmailAttributes.java [hunk 1]
```diff
@@ -207,7 +208,7 @@ public LocalDateTime getSentDate() {
    * @return all the headers of this email message.
    */
   public Map<String, String> getHeaders() {
-    return headers;
+    return headers != null ? ImmutableMap.copyOf(headers) : emptyMap();
```
Comment (alegmarra) [inline extensions/email/src/main/java/org/mule/extension/email/api/EmailAttributes.java]: emptyMap => ImmutableMap.of()

File: extensions/email/src/main/java/org/mule/extension/email/internal/sender/SenderOperations.java [hunk 2]
```diff
@@ -45,41 +44,35 @@
...
                    @DisplayName("Additional Headers") @Optional Map<String, String> headers,
                    @Optional List<EmailAttachment> attachments) {
-    sendOperation.send(connection, content, subject, toAddresses, configuration.getFrom(), configuration.getDefaultCharset(),
+    sendOperation.send(connection,
+                       content,
+                       subject,
+                       toAddresses,
+                       configuration.getFrom(),
+                       configuration.getDefaultCharset(),
                        ccAddresses != null ? ccAddresses : new ArrayList<>(),
```
Comment (alegmarra) [inline extensions/email/src/main/java/org/mule/extension/email/internal/sender/SenderOperations.java]: new ArrayList<>() => emptyList

File: extensions/email/src/main/java/org/mule/extension/email/internal/sender/SenderOperations.java [hunk 3]
```diff
@@ -45,41 +44,35 @@
...
-                   // support
-                   // for
-                   // it
-                   // -
-                   // MULE-9181.
-                   @Optional(defaultValue = "[No Subject]") String subject, List<String> toAddresses,
-                   @Optional List<String> ccAddresses, @Optional List<String> bccAddresses,
+  public void send(@Connection SenderConnection connection,
+                   @UseConfig SMTPConfiguration configuration,
+                   EmailContent content, // TODO: create transformer from string to EmailContentMULE-9181.
```
Comment (marianogonzalez) [inline extensions/email/src/main/java/org/mule/extension/email/internal/sender/SenderOperations.java]: do not use trailing comments.

File: extensions/email/src/main/java/org/mule/extension/email/api/EmailAttributes.java [hunk 4]
```diff
@@ -208,7 +207,7 @@ public LocalDateTime getSentDate() {
    * @return all the headers of this email message.
    */
   public Map<String, String> getHeaders() {
-    return headers != null ? ImmutableMap.copyOf(headers) : emptyMap();
+    return headers != null ? ImmutableMap.copyOf(headers) : ImmutableMap.of();
```
Comment (marianogonzalez) [inline extensions/email/src/main/java/org/mule/extension/email/api/EmailAttributes.java]: rollback. This class is API. Cannot depend on an unexported dependency

**Pull request merged**